python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -0,0 +1,741 @@
+        ),
+        "uint8": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.uint8())
+        ),
+        "uint16": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.uint16())
+        ),
+        "uint32": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.uint32())
+        ),
+        "uint64": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.uint64())
+        ),
+        "float32": plc.interop.from_arrow(
+            pa.array([1.0, None, 3.0, None], type=pa.float32())
+        ),
+        "float64": plc.interop.from_arrow(
+            pa.array([1.0, 2.0, None, None], type=pa.float64())
+        ),
+        "object": plc.interop.from_arrow(
+            pa.array(["a", None, "c", None], type=pa.string())
+        ),
+        "bool": plc.interop.from_arrow(
+            pa.array([True, False, None, None], type=pa.bool_())
+        ),
+        "datetime64[ns]": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.timestamp("ns"))
+        ),
+        "datetime64[ms]": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.timestamp("ms"))
+        ),
+        "datetime64[us]": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.timestamp("us"))
+        ),
+        "datetime64[s]": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.timestamp("s"))
+        ),
+        "timedelta64[ns]": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.duration("ns"))
+        ),
+        "timedelta64[ms]": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.duration("ms"))
+        ),
+        "timedelta64[us]": plc.interop.from_arrow(
+            pa.array([1, None, 3, None], type=pa.duration("us"))
+        ),
+        "timedelta64[s]": plc.interop.from_arrow(
+            pa.array([1, 2, None, None], type=pa.duration("s"))
+        ),
+    }
+
+
+@pytest.fixture(scope="module")
+def null_columns(columns):
+    breakpoint()
+
+
+@pytest.fixture(scope="module", params=LIBCUDF_SUPPORTED_TYPES)
+def binop_lhs_ty(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=LIBCUDF_SUPPORTED_TYPES)
+def binop_rhs_ty(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=LIBCUDF_SUPPORTED_TYPES)
+def binop_out_ty(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=BINARY_OPS,
+)
+def binary_operator(request):
+    return request.param
+
+
+def make_test(binop_lhs_ty, binop_rhs_ty, binop_out_ty, binary_operator):
+    fail = False
+    if not plc.binaryop._is_supported_binaryop(
+        dtype_to_pylibcudf_type(binop_out_ty),
+        dtype_to_pylibcudf_type(binop_lhs_ty),
+        dtype_to_pylibcudf_type(binop_rhs_ty),
+        binary_operator,
+    ):
+        fail = True
+    return (binop_lhs_ty, binop_rhs_ty, binop_out_ty, fail)
```

Comment:
I think if you use my suggestion above about parametrizing by typed columns instead of by types indexing into columns, it'll also make sense to move the failure detection into `_test_binaryop_inner`. Then you won't need this helper function at all, nor would you require the indexing logic into `columns` in `_test_binaryop_inner`.